bin/inx-chronicle/src/shutdown.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use log::warn;
+use tokio::sync::oneshot;
+
+pub(crate) type ShutdownTx = oneshot::Sender<()>;
+pub(crate) type ShutdownRx = oneshot::Receiver<()>;
+
+#[cfg(all(unix, feature = "docker"))]
+pub(crate) use tokio::signal::unix::SignalKind;
+
+/// Creates a shutdown listener for Unix platforms.
+#[cfg(all(unix, feature = "docker"))]
+pub(crate) fn shutdown_listener(signals: Vec<SignalKind>) -> ShutdownRx {
+    use futures::future;
+    use tokio::signal::unix::{signal, Signal};
+
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
+
+    tokio::spawn(async move {
+        let mut signals = signals
+            .iter()
+            .map(|kind| signal(*kind).unwrap())
+            .collect::<Vec<Signal>>();
+
+        let signal_futures = signals.iter_mut().map(|signal| Box::pin(signal.recv()));
+
+        let (signal_event, _, _) = future::select_all(signal_futures).await;
+
+        if signal_event.is_none() {
+            panic!("Shutdown signal stream failed, channel may have closed.");
+        } else {
+            shutdown_procedure(shutdown_tx);
+        }
+    });
+
+    shutdown_rx
+}
+
+/// Creates a shutdown listener for Non-Unix platforms.
+#[cfg(not(all(unix, feature = "docker")))]
```

Comment:
These should be `#[cfg(all(not(unix), feature = "docker"))]`

Reply:
Sure? This has to be the negation of the above `#[cfg(all(unix, feature = "docker"))]` to really be complementary, and what you propose is not the logical negation.

Reply:
You're right, I was assuming there would be a third feature set but this is the old impl.